internal/pkg/aead/aead_test.go

```diff
@@ -89,3 +93,72 @@ func TestMarshalAndUnmarshalStruct(t *testing.T) {
 		t.Fatalf("expected structs to be equal")
 	}
 }
+
+func TestCipherDataRace(t *testing.T) {
```

Comment:
Consider annotating with a comment pointing to a GitHub issue describing the problem. Six months from now, we'll all forget why we added this test :-)

Reply:
Added